pkg/cortex/cortex_test.go

```diff
@@ -94,3 +94,46 @@ func TestCortex(t *testing.T) {
 	// check that compactor is configured which is not part of Target=All
 	require.NotNil(t, serviceMap[Compactor])
 }
+
+func TestConfigValidation(t *testing.T) {
+	for _, tc := range []struct {
+		name          string
+		getTestConfig func() *Config
+		expectedError bool
```

Comment:
To have an higher confidence on test correctness, you could check the exact error. We expect `errInvalidHttpPrefix` (see other comment).

```suggestion
		expectedError error
```